sorterTests.js

```diff
@@ -0,0 +1,161 @@
+                            propertiesToTest[property].supported = false
+                        }
+                    } else {
+                        // If asc item is not less than desc item, then sorter didn't work.  This checks numbers and booleans.
+                        if (ascItem > descItem) {
+                            propertiesToTest[property].supported = false
+                        }
+                    }
+                }
+            }
+        } catch (error) {
+            propertiesToTest[property].supported = false
+        }
+    }
+
+    return propertiesToTest
+}
+
+async function validateSorters(httpClient, method, version, path, spec) {
+    let uniqueErrors = {
+        method: method,
+        endpoint: version + path,
+        errors: {
+            undocumentedSorters: [],
+            unsupportedSorters: []
+        }
+    };
+    let documentedSorters = null;
+
+    if (spec.paths[path].get.parameters != undefined) {
+        const filteredParams = spec.paths[path].get.parameters.filter(param => param.name === "sorters")
+        const schema = spec.paths[path].get.responses['200'].content['application/json'].schema;
+        if (filteredParams.length == 1) {
+            try {
+                documentedSorters = parseSorters(filteredParams[0].description);
+                const propertiesToTest = getSortableProperties(schema)
+                const testedProperties = await testSorters(httpClient, path, propertiesToTest, documentedSorters)
+                for (property in testedProperties) {
+                    // If the property wasn't able to be tested (ex. not enough data), then don't report any errors
+                    if (testedProperties[property].testable) {
+                        if (documentedSorters.includes(property)) {
+                            if (!testedProperties[property].supported) {
+                                uniqueErrors.errors['unsupportedSorters'].push({
+                                    'message': `The property \`${property}\` **MIGHT NOT** support sorting but the documentation says it does. Please manually verify.`,
+                                    'data': null
+                                })
+                            }
+                        } else {
+                            if (testedProperties[property].supported) {
+                                uniqueErrors.errors['undocumentedSorters'].push({
+                                    'message': `The property \`${property}\` **MIGHT** support sorting but it is not documented. Please manually verify.`,
+                                    'data': null
+                                })
+                            }
+                        }
+                    }
+                }
+            } catch (error) {
+                uniqueErrors.errors['Invalid Sorters'] = {
+                    'message': `Unable to parse the sorters due to improper format: ${error.message}`,
```

Comment:
You are correct, but I offloaded the formatting error messages to the linter since the linter deals with static code analysis while the schema validator deals with dynamic analysis.  You can find all of the error messages that would be produced due to bad formatting in the tests for the linter:

- https://github.com/sailpoint-oss/api-linter/blob/main/functions/tests/parameter-filter-description-check-test.js
- https://github.com/sailpoint-oss/api-linter/blob/main/functions/tests/parameter-sorter-description-check-test.js

Reply:
Yes that's fine, so as long as the user will know exactly what the format needs to be along with the error